src/adjoint_common.jl

```diff
@@ -427,13 +427,8 @@ function ReverseLossCallback(sensefun, λ, t, dgdu, dgdp, cur_time)
     @unpack factorized_mass_matrix = sensefun.diffcache
     prob = getprob(sensefun)
     idx = length(prob.u0)
-    if ArrayInterfaceCore.ismutable(y)
-        return ReverseLossCallback(isq, λ, t, y, cur_time, idx, factorized_mass_matrix,
-                                   sensealg, dgdu, dgdp, sensefun.diffcache, nothing)
-    else
-        return ReverseLossCallback(isq, λ, t, y, cur_time, idx, factorized_mass_matrix,
-                                   sensealg, dgdu, dgdp, sensefun.diffcache, sensefun.sol)
-    end
+    return ReverseLossCallback(isq, λ, t, y, cur_time, idx, factorized_mass_matrix,
+                                sensealg, dgdu, dgdp, sensefun.diffcache, sensefun.sol)
```

Comment:
```suggestion
                                sensealg, dgdu, dgdp, sensefun.diffcache, sensefun.f)
```

Reply:
oh wait hmm, some cases need `sol`. This should be algorithm dependent, and the `f` should be kept separate.